### PR TITLE
Add `sha256_hash` ABI

### LINKED ISSUE #3491: Add "sha256" ABI
Add a "sha256" ABI taking bytes as input, computing the sha256 of the bytes, and giving the hash as output bytes.

### PR DESCRIPTION
**Pull request by aoudiamoncef**
* [x] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

closes #3491

### CODE REVIEW

**Review by AurelienFT — CHANGES_REQUESTED** [review]

File: massa-execution-worker/src/interface_impl.rs [hunk 1]
```diff
@@ -761,4 +762,18 @@ impl Interface for InterfaceImpl {
...
     }
+
+    /// Hashes givens byte array with sha256
+    ///
+    /// # Arguments
+    /// * bytes: byte array to hash
+    ///
+    /// # Returns
+    /// The vector of bytes representation of the resulting hash
+    fn hash_sha256(&self, bytes: &[u8]) -> Result<[u8; 32]> {
```
[the remark this answers is not in the document]
Comment (aoudiamoncef) [reply massa-execution-worker/src/interface_impl.rs:773]: It was planned but I canceled It as there is no test for ABI's in massa core. May be it's time to create a dedicated tests file ?

[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-execution-worker/src/interface_impl.rs:773]: Yes you can introduce some tests on the interface massa side

File: massa-node/base_config/gas_costs/abi_gas_costs.json [hunk 2]
```diff
@@ -45,5 +45,6 @@
   "assembly_script_transfer_coins": 176,
   "assembly_script_transfer_coins_for": 208,
   "assembly_script_unsafe_random": 145,
+  "assembly_script_hash_sha256": 244,
```
[the remark this answers is not in the document]
Comment (aoudiamoncef) [reply massa-node/base_config/gas_costs/abi_gas_costs.json:48]: Cool, I used the same value as the existing hashing function. 

Take a look at https://github.com/massalabs/massa-sc-runtime/issues/228

File: massa-execution-worker/src/interface_impl.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (aoudiamoncef) [reply massa-execution-worker/src/interface_impl.rs:773]: The test is trivial but I found the integration painful

```rust
#[test]
#[serial]
fn test_hash_sha256() {
	//THIS IS AVAILABLE ONLY in #[cfg(any(feature = "gas_calibration", feature = "benchmarking"))]
    let interface = InterfaceImpl::new_default(
        Address::from_str("AU12cMW9zRKFDS43Z2W88VCmdQFxmHjAo54XvuVV34UzJeXRLXW9M").unwrap(),
        None,
    );
    let actual_hash = interface.hash_sha256("something".as_bytes()).unwrap();
    let expected_hash =
        "3fc9b689459d738f8c88a3a48aa9e33542016b7a4052e001aaa536fca74813cb".as_bytes();
    assert_eq!(actual_hash, expected_hash);
}
```

If you have a better idea, I'll take it

[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-execution-worker/src/interface_impl.rs:773]: You can add the testing feature to the list of feature gas_calibration and benchmarking

File: massa-node/base_config/gas_costs/abi_gas_costs.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-node/base_config/gas_costs/abi_gas_costs.json:48]: I need to have this merged on testnet to run the gas-calib. Can you remove this change ? And I will fix the CI on testnet branch using our bot

File: massa-execution-worker/src/interface_impl.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (aoudiamoncef) [reply massa-execution-worker/src/interface_impl.rs:781]: thanks for the boost 👍🏻 . Tests are working let's gooo

**Review by AurelienFT — COMMENTED** [review]
Can you fix this two clippy lints ? https://github.com/massalabs/massa/actions/runs/4173498451/jobs/7226016090 and we are good to go for merging !

**Comment by aoudiamoncef** [comment]
> Can you fix this two clippy lints ? [massalabs/massa/actions/runs/4173498451/jobs/7226016090](https://github.com/massalabs/massa/actions/runs/4173498451/jobs/7226016090) and we are good to go for merging !

I dont have any clippy smells with: `rustc 1.69.0-nightly (e972bc808 2023-01-29) && clippy 0.1.69 (e972bc80 2023-01-29)`

**Comment by aoudiamoncef** [comment]
@AurelienFT it's valentine gift 💟 . the toolchain is up to date.

**Review by AurelienFT — APPROVED** [review]
Nice one !

**Pull request merged**